src/bootstrap/bootstrap.py

```diff
@@ -1162,6 +1162,49 @@ def build_triple(self):
         config = self.get_toml("build")
         return config or default_build_triple(self.verbose)
 
+    def is_git_repository(self, repo_path):
+        return (
+            subprocess.run(
+                ["git", "-C", repo_path, "rev-parse", "--is-inside-work-tree"],
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            ).returncode
+            == 0
+        )
+
+    def get_latest_commit(self, repo_path, branch, author_email):
+        try:
+            if not self.is_git_repository(repo_path):
+                return "<commit>"
+            cmd = [
+                "git",
+                "-C",
+                repo_path,
+                "log",
+                "--format=%H",
+                "--author",
+                author_email,
+                branch,
+                "-n",
+                "1",
+            ]
```

Comment:
To get proper nightly commit-finding logic, we ideally need to replicate how bootstrap finds commits for ci-rustc. However, I plan to remove the entire python source from the bootstrap soon (after [this PR](https://github.com/rust-lang/rust/pull/119899)), so it's not worth putting too much effort here. You can replace this with `git rev-list --author=$author -n1 HEAD` and it should be fine for this task.